.github/workflows/on-merge-to-main.yml

```diff
@@ -18,6 +18,8 @@ jobs:
     secrets:
       token: ${{ secrets.GITHUB_TOKEN }}
   run-unit-tests:
+    needs: get_pr_details
+    if: ${{ needs.get_pr_details.outputs.prIsMerged == 'true' }}
```

Comment:
Q: why do we need `prIsMerged` to be true? I thought this workflow is triggerred only when PR merged.

Reply:
This workflow is actually triggered by [`record_pr.yml`](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.github/workflows/record_pr.yml#L5), which in turn is triggered every time a PR is opened, updated (title or description change), or closed. This is a design that we copied from the Powertools Python repo.

The reason why we don't want to trigger directly is that this workflow is somewhat privileged and we don't want to allow forks to access it. So we trigger the workflow that exports the PR details for most events, and then dispatch the other downstream workflows. Then those workflows can decide what to do based on the content/status of the PR.